src/checkout.js

```diff
@@ -0,0 +1,150 @@
+
+    this.checkout_button_sub = Checkout.replaceCheckoutButton();
+    this.checkout_button_sub.addEventListener(
+      "click",
+      this.checkoutButtonClicked
+    );
+
+    this.dialog = Checkout.createDialog();
+
+    this.dialog
+      .querySelector("#yes")
+      .addEventListener("click", this.yesButtonClicked);
+
+    this.dialog
+      .querySelector("#not_now")
+      .addEventListener("click", this.closeDialogAndGoToCart);
+
+    this.dialog
+      .querySelector("#no")
+      .addEventListener("click", this.noButtonClicked);
+
+    this.dialog.querySelector("#bes_close").addEventListener("click", () => {
+      this.dialog.style.display = "none";
+    });
+
+    this.port.onMessage.addListener(this.applyConfig);
+    this.port.postMessage({ requestConfig: {} }); // TO DO: this must be at end of init
+  }
+
+  static applyConfig(msg) {
+    this.log.info("config recieved from backend" + JSON.stringify(msg.config));
+    this.config = msg.config;
+  }
+
+  static checkoutButtonClicked() {
+    if (
+      !this.config.albumOnCheckoutDisabled &&
+      !this.config.albumPurchasedDuringCheckout
+    ) {
+      this.dialog.style.display = "block";
+      return;
+    }
+
+    this.closeDialogAndGoToCart();
+  }
+
+  static closeDialogAndGoToCart() {
+    this.dialog.style.display = "none";
+
+    const clickEvent = new MouseEvent("click", {
+      view: window,
+      bubbles: true,
+      cancelable: false
+    });
+    let checkout_button = document.querySelector("#sidecartCheckout");
+    checkout_button.dispatchEvent(clickEvent);
+  }
+
+  static yesButtonClicked() {
+    let unit_price = this.dialog.querySelector("input").value;
+    this.log.info(`price ${unit_price}`);
+
+    let error_div = this.dialog.querySelector("#bes_checkout_error");
+    error_div.innerHTML = " ";
+    if (unit_price < 5) {
+      error_div.innerHTML = "value entered is under $5.00 CAD";
+      return;
+    }
+
+    this.log.info(`add ${albumsToPurchase} to cart`);
+    this.addAlbumToCart("a", albumsToPurchase, unit_price)
+      .then(() => {
+        this.log.info("update config");
+        this.port.postMessage({
+          config: { albumPurchasedDuringCheckout: true }
+        });
+      })
+      .finally(() => {
+        this.closeDialogAndGoToCart();
+      });
+  }
+
+  static noButtonClicked() {
+    this.port.postMessage({ config: { albumOnCheckoutDisabled: true } });
+    this.closeDialogAndGoToCart();
+  }
+
+  static replaceCheckoutButton() {
+    let checkout_button = document.querySelector("#sidecartCheckout");
+    checkout_button.style.display = "none";
+
+    let sidecart_footer = document.querySelector("#sidecartFooter");
+    let checkout_button_sub = document.createElement("a");
+    checkout_button_sub.className = "buttonLink notSkinnable";
+    checkout_button_sub.innerHTML = checkout_button.innerHTML;
+
+    sidecart_footer.appendChild(checkout_button_sub);
+
+    return checkout_button_sub;
+  }
```

Comment:
I'm curious about the chances of failures after the original checkout button is hidden. Is it possible for the use to be left without a checkout button at all? Is it worth rearranging this method so that all the querySelectors are executed before the DOM is manipulated?

Reply:
sounds fair to me